configuration/reports/reportdescriptors/dataexports/sql/j9CaseRegistration.sql

```diff
@@ -44,27 +47,47 @@ where pp.patient_program_id =
          	and pp2.program_id = @matHealthProgram 
          	and pp2.voided = 0  
          	and pp2.date_completed is null
-         	order by pp2.date_enrolled desc limit 1)
-;
+         	order by pp2.date_enrolled desc limit 1);
+         	
+-- load one row per patient in program
+insert into temp_J9_patients (patient_id, patient_program_id, j9_enrollment_date)
+Select pp.patient_id, pp.patient_program_id ,pp.date_enrolled 
+FROM temp_mch_pp pp 
+WHERE  pp.program_id = @matHealthProgram and pp.voided = 0  and pp.date_completed is NULL;
 
--- this populates the temp table with the cohort of patients to be included in the report
--- The logic is, patients that:
--- * have ever been in the maternal health program
--- * maximum of one row per patient.  So return the most recent enrollment if more than one
-insert into temp_J9_patients (patient_id, patient_uuid, j9_enrollment_date, J9_program ,birthdate, age)
-Select p.patient_id, per.uuid, pp.date_enrolled, currentProgramState(pp.patient_program_id,@MCH_treatment,'en'),
-per.birthdate, TIMESTAMPDIFF(YEAR,per.birthdate, now()) "age"
-from patient p
-         inner join person per on per.person_id = p.patient_id and per.dead = 0
-         inner join temp_mch_pp pp on pp.patient_id = p.patient_id 
-;
+create index temp_J9_patients_pi on temp_J9_patients(patient_id);
+
+-- delete patients who are dead
+delete t from temp_J9_patients t
+inner join person p on p.person_id = t.patient_id
+where p.dead = 1;
+
+-- load all relevant encounters (for use later to narrow down observations)
```

Comment:
Shouldn't this table be called temp_j9_enc or something, since it includes both obgyn and delivery encounters?  Also, can we move this further down the query right before it is actually used, so it's easier to follow the progression of thought?